### DISCUSSION TITLE
Replacing editorjs

### DISCUSSION THREAD

**Discussion by allen-munsch** [mirumee/saleor-dashboard#5221] [RFC / Ideas] [answered]
I had a very difficult time using the dashboards content/page creation text editor.

I found it easier to write a graphql mutation out of the curl request from the network dev tools, and having an LLM write for me a bash script, which would convert the markdown into EditorJS format.

EditorJS doesn't support markdown, html, or other formats which are very commonly used out of the box in other editors, so copy pasting, of those other formats causes a lot of friction around content management.

In addition to this the basic components / blocks give me PTSD of having to use Notion, for me Notion was a terrible experience, and almost as bad as using Jira/Confluence.

Related:

- https://github.com/codex-team/editor.js/discussions/1870 
- https://github.com/saleor/saleor-dashboard/discussions/2141 

Possible Alternatives:

- https://github.com/ckeditor/ckeditor5/blob/d9d69bacc6717321bf044ff4d047eb0f8e2e4a33/packages/ckeditor5-markdown-gfm/src/markdown.ts#L18 
- https://ui.toast.com/tui-editor

I do know that the editorjs field data is saved in the backend, so I am not sure how much of a lift it would be to replace it.

**Comment by lkostrowski** [comment, marked as answer]
Hey @allen-munsch 

thanks for your feedback. We are aware that using Editor.js is not a pleasant experience.

We will eventually look for the alternative, but to make the right choice we can't rush with that - as you mentioned it's also part of the API. Reasonable approach would be to follow some popular AST standard.

We have included your request into our internal backlog with a feature requests. In the meantime I can suggest connecting external headless CMS platform and keep advanced text there. You can use the CMS app to connect with services like Contentful or Payload

**Reply by allen-munsch** [reply]
Yep, makes sense. Also thanks for the pointers on the CMS app, Brilliant!

Will take a look at: https://github.com/saleor/apps/tree/main/apps/cms-v2
